### PR TITLE
Introduce accessible addon configurations

### LINKED ISSUE #1691: Store and use addon logos and descriptions
**What would you like to be added? / User Story**: We should be able to store logos and descriptions of all accessible addons that are configured or can be configured in our system. This could be done using config maps for example.

After the data will be available we should use in the dashboard to make the feature look like in the designs.

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: Fixes https://github.com/kubermatic/dashboard-v2/issues/1691.

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Introduced addon configurations.
```

### CODE REVIEW

File: api/pkg/crd/kubermatic/v1/addon_config.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// AddonConfigResourceName represents "Resource" defined in Kubernetes
+	AddonConfigResourceName = "addonconfigs"
```
Comment (alvaroaleman) [inline api/pkg/crd/kubermatic/v1/addon_config.go]: All these consts are completely unused or did I miss it?

File: config/kubermatic/crd/crd-addon-configs.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+metadata:
+  name: addonconfigs.kubermatic.k8s.io
+spec:
+  group: kubermatic.k8s.io
+  names:
+    kind: AddonConfig
+    listKind: AddonConfigList
+    plural: addonconfigs
+    singular: addonconfig
+  scope: Cluster
```
Comment (alvaroaleman) [inline config/kubermatic/crd/crd-addon-configs.yaml:12]: Addons are namespaced, so wouldnt it be better to also make the `addonconfigs` namespaced?

Comment (maciaszczykm) [reply config/kubermatic/crd/crd-addon-configs.yaml:12]: Addon configs are global, there will be only 1 config per accessible addon. Addons can differ for each cluster, depending on the user's choice.

File: api/pkg/crd/kubermatic/v1/addon_config.go (hunk 1 above)
Comment (maciaszczykm) [reply api/pkg/crd/kubermatic/v1/addon_config.go]: Yes, I will remove them for now.

**Comment by maciaszczykm** [comment]
@alvaroaleman PTAL

**Comment by alvaroaleman** [comment]
/hold
```
The CustomResourceDefinition "addonconfigs.kubermatic.k8s.io" is invalid: spec.scope: Invalid value: "Cluster": field is immutable
```

That's a valid failure.

**Comment by maciaszczykm** [comment]
/hold cancel

The issue was caused by the old version of CRD deployed by the previous check, with different scope. I have removed the CRD from the CI cluster and it should pass right now.

**Pull request merged**